stubs/requests/requests/sessions.pyi

```diff
@@ -49,8 +49,8 @@ _Hook = Callable[[Response], Any]
 _Hooks = MutableMapping[Text, List[_Hook]]
 _HooksInput = MutableMapping[Text, Union[Iterable[_Hook], _Hook]]
 
-_ParamsMappingKeyType = Union[Text, bytes, int, float]
-_ParamsMappingValueType = Union[Text, bytes, int, float, Iterable[Union[Text, bytes, int, float]], None]
+_ParamsMappingKeyType = Union[Text, bytes, int, float, bool]
```

Comment:
Passing in `True` as a key is more likely to be a mistake than something you intended, so I don't think we should support that. Can you think of a situation where `{True: "lol"}` would actually be useful?

Reply:
I could contrive an example, but I don't think that's particularly useful. IMHO, type stubs exist to help annotate what is possible, not what is best practices. For better or worse, passing a boolean "works" in that `requests` uses it in the request.

Reply:
I don't agree with that in general: there are lots of things that "work" at runtime but probably indicate a bug, and type checking is intended to help catch those. For example, almost any boolean function parameter will work with any type in practice, but we still allocate them as `bool` in typeshed.

So I'd like to see a concrete example of something that this PR fixes before we merge it.

Reply:
In this PR case, I'm coming to the conclusion that my test setup (which was a bit abtracted) had something messy in it and I'm happy to close this PR for now since indeed this should be covered by `int`.

That said, I'm still not happy with this type setup here for Param and I think I'll come back with a better PR to fix these types. I do indeed have some logic and examples for those cases, but it'll take me a while to organize that. Thanks for the conversation however!